bridges/modules/xcm-bridge-hub-router/src/lib.rs

```diff
@@ -408,12 +408,12 @@ impl<T: Config<I>, I: 'static> SendXcm for Pallet<T, I> {
 }
 
 impl<T: Config<I>, I: 'static> InspectMessageQueues for Pallet<T, I> {
-	fn clear_messages() {
-		ViaBridgeHubExporter::<T, I>::clear_messages()
-	}
+	fn clear_messages() {}
 
+	/// This router needs to implement `InspectMessageQueues` but doesn't have to
+	/// return any messages, since it just reuses the `XcmpQueue` router.
 	fn get_messages() -> Vec<(VersionedLocation, Vec<VersionedXcm<()>>)> {
-		ViaBridgeHubExporter::<T, I>::get_messages()
+		Vec::new()
```

Comment:
Clearing messages after returning wouldn't change anything, since after the runtime API storage won't be affected.
The reason I changed this to an empty vec is because this was calling the internal `get_messages` and that internal type was `XcmpQueue`, a type that already returns the messages correctly. So this implementation was just duplicating messages in the response.

Say we dry run a call that ends up sending a message to parachain X. The router for the asset hubs is set as a tuple with three routers `(XcmpQueue, Exporter1, Exporter2)`. The tuple implementation will go type by type. The first type already returns the message that would be sent to parachain X. The second was calling `get_messages` internally to `XcmpQueue` again, so it would append the exact same message. Same for the third implementation. Because of this, we end up with the same message repeated 3 times.
After looking at these I concluded that even though these types appear in the router, they shouldn't report their own messages, they are already included in the queues of `XcmpQueue`.

Reply:
Ok. Thank you for the details !